Agent assist verify (#124)
watson-developer-cloud/assistant-toolkit: integrations/webchat/agent-assist/agent-verify/README.md

### Step 2: Create a new queue

![Genesys Queue Page](images/create-queue-page.png)

Click create queue and specify a name for the new queue. Read about queue options [here](https://help.mypurecloud.com/articles/create-queues).

### Step 3: Configure the queue

![Genesys Queue Page](images/queue-config-page.png)

After specifying the members or groups associated with the new queue, you can simply take the defaults for the queue configuration with the exception of the `Voice Transcription` field outlined in red in the above image which needs to be set to On.

### Step 4: Go to the Integrations page from the Admin panel

![Genesys Admin Page](images/admin-page.png)

From the Admin panel click on the `Integrations` link to create a new integration.

### Step 5: Create a new integration

![Genesys Integrations Page](images/integrations-page.png)

Add a new integration by clicking the + sign in the top right of the page.

### Step 6: Install the Interaction Widget

![Genesys Interaction Widget Page](images/interaction-widget-page.png)

The Interaction Widget is needed to embed third-party webapps via an iframe in the Agent UI.

> ### Step 7: Configure the Interaction Widget

Reviewer: I would reverse steps seven and eight. Because you need the client ID for the interaction widget, I would set that up first.

`conversationID`: generated by Genesys at runtime and the field will be populated by the agent application.

#### Iframe Sandbox Options
The `Iframe Sandbox Options` should be filled in with these options:

```
allow-scripts,allow-same-origin,allow-forms,allow-modals,allow-popups,allow-presentation,allow-downloads
```
#### Group Filtering
The `Group Filtering` section is selected from a pre-configured group that can be created from the Admin page under Directory->Groups. You need to make sure that all agents that require the new agent-verify widget are listed within the chosen group.

### Step 8: Configure OAuth for the Interaction Widget

![Genesys Interaction Widget Config Page](images/oauth-client-page.png)
Start by going to `OAuth` under Integrations and it `Add Client` to create a new OAuth client.

![Genesys Interaction Widget Config Page](images/oauth-client-details-top-page.png)
Here you'll need to configure the App Name associated with the credentials, the Grant Type which must be `Token Implicit Grant (Browser)`, the redirect URL which is shown below:
```
http://localhost:8080/genesys-agent-assist-verify.html?clientID=xxxxx&environment=mypurecloud.com
```

> and the Scope which is set to `notifications`. Note that the redirect URL does not need to include the conversationID.

Reviewer: "does not need" really should be "must not contain". You cannot put the conversation ID in the redirect URL. I have discovered the redirect URLs must match exactly, including any query parameters.

Reviewer: FYI, one trick you can use to hide sensitive information in your screenshots is that when you're using the development tools in your web browser you can actually directly edit the HTML. So you could just edit the HTML that shows the client secret and fill in some bogus data and then you don't have to try to mask it in the screenshot.

![Genesys Interaction Widget Config Page](images/oauth-client-details-bottom-page.png)
Once the previous page is saved, you will be presented with more details about your OAuth configuration including the Client ID which is needed when configuring the Interaction Widget.

### Step 9: Setup Speech and Text Analytics

![Genesys Speech Analytics Page](images/speech-analytics-page.png)

Navigate to the `Speech and Text Analytics` page from the Admin page.

![Genesys Speech Analytics Config Page](images/speech-analytics-config-page.png)

> On this page you will need to enable voice transcriptions and `Low Latency Transcription`. You can read more about the settings on this page [here](https://help.mypurecloud.com/articles/speech-and-text-analytics).

Reviewer: FYI, it is considered bad practice for accessibility to use "here" links. Basically the idea is that if the user selects the link with their screenreader, they don't really know what it does because the text just says "here". The link should really be the text of where it is going to.
```suggestion
On this page you will need to enable voice transcriptions and `Low Latency Transcription`. You can read the [speech setting documentation](https://help.mypurecloud.com/articles/speech-and-text-analytics) for more information.
```

----

After completing these steps you will need to setup a call flow to test that the agent-verify widget is being loaded into the Agent Desktop. Here are a few links to help you get started:

- [Integrating Watson Assistant with phone and Genesys Cloud](https://cloud.ibm.com/docs/watson-assistant?topic=watson-assistant-deploy-phone-genesys)
- [Genesys: Create a trunk under BYOC Cloud](https://help.mypurecloud.com/articles/create-a-byoc-cloud-trunk/)
- [About Genesys Cloud Voice](https://help.mypurecloud.com/articles/about-genesys-cloud-voice/)
- [Details on Genesys Voice Transcription](https://help.mypurecloud.com/articles/about-voice-transcription/)

## Miscelaneous

> This sample relies on the `string-similarity` algorithm implemented [here](https://github.com/aceakash/string-similarity) to do the verbatim check.

Reviewer: ```suggestion
This sample relies the [string-similarity](https://github.com/aceakash/string-similarity) library to do the verbatim check.
```
